### PR TITLE
Cleanup Docker build + use docker-compose for test

### PR DESCRIPTION
**Pull request by tringuyen-yw**
- Cleanup Docker build, modeled on [samples-js-shoppingcart](https://github.com/cloudstateio/samples-js-shoppingcart/tree/master/shopping-cart)
- Use `docker-compose` to unit test the gRPC service (instead of starting individually the docker images of CS proxy and the service)
- Use tag `0.5.1` for cloudstateio/cloudstate-proxy-dev-mode

### CODE REVIEW

File: friends/package.json [hunk 1]
```diff
@@ -28,13 +28,15 @@
   },
   "dependencies": {
     "minimist": ">=0.2.1",
-    "cloudstate": "0.0.3"
+    "cloudstate": "0.0.3",
+    "@grpc/proto-loader": "^0.1.0",
+    "google-protobuf": "^3.0.0",
+    "grpc": "^1.20.2"
```
Comment (vkorenev) [inline friends/package.json]: `package-lock.json` should be updated and put to git as well.

File: friends/package.json [hunk 2]
```diff
@@ -28,13 +28,15 @@
...
   },
   "scripts": {
-    "prestart": "compile-descriptor friends.proto",
+    "build": "compile-descriptor ./friends.proto",
     "start": "node index.js",
-    "start-no-prestart": "node index.js",
-    "dockerbuild": "docker build -t ${DOCKER_PUBLISH_TO:-cloudstateio}/samples-js-chat-friends:latest .",
+    "dockerbuild": "docker build -f ./Dockerfile -t ${DOCKER_PUBLISH_TO:-cloudstateio}/samples-js-chat-friends:latest ../..",
```
Comment (vkorenev) [inline friends/package.json]: `-f Dockerfile` can be replaced by `.`

Comment (vkorenev) [reply friends/package.json]: What is the last `../..` for?

File: presence/package.json [hunk 3]
```diff
@@ -28,13 +28,15 @@
   },
   "dependencies": {
     "minimist": ">=0.2.1",
-    "cloudstate": "0.0.3"
+    "cloudstate": "0.0.3",
+    "@grpc/proto-loader": "^0.1.0",
+    "google-protobuf": "^3.0.0",
+    "grpc": "^1.20.2"
```
Comment (vkorenev) [inline presence/package.json]: `package-lock.json` should be updated and put to git as well.

**Review by vkorenev — CHANGES_REQUESTED** [review]

**Comment by vkorenev** [comment]
I think it is better to have only one `docker-compose.yml` which runs all the services.

File: friends/package.json (hunk 2 above)
Comment (tringuyen-yw) [reply friends/package.json]: I'd like to know why too, took it from [samples-js-shoppingcart/shopping-cart/package.json](https://github.com/cloudstateio/samples-js-shoppingcart/blob/master/shopping-cart/package.json#L45)

**Comment by tringuyen-yw** [comment]
> I think it is better to have only one `docker-compose.yml` which runs all the services.

Good idea, let me try.

File: presence/package.json [hunk 4]
```diff
@@ -28,14 +28,13 @@
...
```
Comment (vkorenev) [inline presence/package.json]: I think we decided to remove these packages from explicit dependencies since the `cloudsate` package depends on them.

**Review by vkorenev — APPROVED** [review]

File: presence/package.json (hunk 4 above)
Comment (tringuyen-yw) [reply presence/package.json]: Grrr, good catch

**Review by vkorenev — APPROVED** [review]

**Pull request merged**